libs/common/Memory/LimitedFixedBufferPool.cs

```diff
@@ -107,59 +134,78 @@ public void Dispose()
 #if HANGDETECT
             int count = 0;
 #endif
-            while (totalAllocations > int.MinValue &&
-                Interlocked.CompareExchange(ref totalAllocations, int.MinValue, 0) != 0)
+            while (totalReferences > int.MinValue &&
+                Interlocked.CompareExchange(ref totalReferences, int.MinValue, 0) != 0)
             {
 #if HANGDETECT
                     if (++count % 10000 == 0)
                         logger?.LogTrace("Dispose iteration {count}, {activeHandlerCount}", count, activeHandlerCount);
 #endif
                 Thread.Yield();
             }
-            for (int i = 0; i < numLevels; i++)
+            for (var i = 0; i < numLevels; i++)
             {
                 if (pool[i] == null) continue;
                 while (pool[i].size > 0)
```

Comment:
If I'm understanding correctly that .size is total # of allocations which may be > maxEntriesPerLevel, then this can spin forever if there are unReturned items. This might warrant an Assert or logging with an exit. (Holding an unReturned item and calling Dispose() (and Purge()?) is bad anyway, so let's make it easier to catch).

Reply:
This PR does not change the LBFP implementation. Changes to its core operations should be part of a separate PR.